hetu-filesystem-client/src/main/java/io/hetu/core/filesystem/HetuLocalFileSystemClient.java

```diff
@@ -216,4 +223,31 @@ public boolean isDirectory(Path path)
     public void close()
     {
     }
+
+    @Override
+    public long getTotalSpace(Path path) throws IOException
+    {
+        FileStore fileStore = getFileStore(path);
+        return fileStore.getTotalSpace();
+    }
+
+    @Override
+    public long getUsableSpace(Path path) throws IOException
+    {
+        FileStore fileStore = getFileStore(path);
+        return fileStore.getUsableSpace();
+    }
+
+    @Override
+    public Path createTemporaryFile(Path path, String prefix, String suffix) throws IOException
+    {
+        return Files.createTempFile(path, prefix, suffix);
+    }
+
+    @Override
+    public Path createTemporaryFileName(Path path, String prefix, String suffix)
+    {
+        String randomNo = String.valueOf(secureRandom.nextLong());
+        return Paths.get(String.valueOf(path), prefix + randomNo + suffix);
```

Comment:
May be random number alone is not sufficient; either use a UUID or prepend WorkerID/Name for keeping it unique or create subfolder by worker name in the HDFS spill path

Reply:
Resolved